src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ITypeSymbolExtensions.TypeSyntaxGeneratorVisitor.cs

```diff
@@ -127,18 +127,56 @@ public static bool TryCreateNativeIntegerType(INamedTypeSymbol symbol, out TypeS
 
 #if !CODE_STYLE
 
+#nullable enable
             public override TypeSyntax VisitFunctionPointerType(IFunctionPointerTypeSymbol symbol)
             {
-                // TODO(https://github.com/dotnet/roslyn/issues/39865): generate the calling convention once exposed through the API
+                FunctionPointerCallingConventionSyntax? callingConventionSyntax = null;
+                if (symbol.Signature.CallingConvention != System.Reflection.Metadata.SignatureCallingConvention.Default)
+                {
+                    SeparatedSyntaxList<FunctionPointerUnmanagedCallingConventionSyntax> conventionsList = default;
+                    switch (symbol.Signature.CallingConvention)
+                    {
+                        case System.Reflection.Metadata.SignatureCallingConvention.CDecl:
+                            conventionsList = SyntaxFactory.SeparatedList(new[] { GetConventionForString("Cdecl") });
```

Comment:
Consider having all of this just assign to an array of FunctionPointerCallingConventionSyntax and then you can put the SyntaxFactory.SeparatedList call in a single place where you're assembling everything.